src/main/java/org/broadinstitute/hellbender/engine/spark/SAMRecordToGATKReadAdapterSerializer.java

```diff
@@ -0,0 +1,97 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import htsjdk.samtools.*;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
+
+public class SAMRecordToGATKReadAdapterSerializer extends Serializer<SAMRecordToGATKReadAdapter> {
+
+    private BAMRecordCodec lazyCodec = new BAMRecordCodec(null, new LazyBAMRecordFactory());
+
+    @Override
+    public void write(Kryo kryo, Output output, SAMRecordToGATKReadAdapter adapter) {
+        SAMRecord record = adapter.getSamRecord();
+        // serialize reference names to avoid having to have a header at read time
+        output.writeString(record.getReferenceName());
+        output.writeString(record.getMateReferenceName());
+        lazyCodec.setOutputStream(output);
+        lazyCodec.encode(record);
+
+        // clear indexing bin after encoding to ensure all SAMRecords compare properly
+        record.setFlags(record.getFlags());
+    }
+
+    @Override
+    public SAMRecordToGATKReadAdapter read(Kryo kryo, Input input, Class<SAMRecordToGATKReadAdapter> type) {
+        final String referenceName = input.readString();
+        final String mateReferenceName = input.readString();
+        lazyCodec.setInputStream(input);
+        final SAMRecord record = lazyCodec.decode();
+
+        // clear indexing bin after decoding to ensure all SAMRecords compare properly
+        record.setFlags(record.getFlags());
+
+        final int referenceIndex = record.getReferenceIndex();
+        final int mateReferenceIndex = record.getMateReferenceIndex();
+        record.setReferenceName(referenceName);
+        record.setMateReferenceName(mateReferenceName);
+
+        // set reference indexes again since setting reference names without a header will unset indexes
+        record.setReferenceIndex(referenceIndex);
+        record.setMateReferenceIndex(mateReferenceIndex);
```

Comment:
This dance is needed due to the fact that we don't have a header so we have to set the reference names and indexes explicitly. This ordering ensures that they all get set safely.